app/src/main/java/org/mozilla/fenix/home/intent/CrashReporterIntentProcessor.kt

```diff
@@ -7,25 +7,25 @@ package org.mozilla.fenix.home.intent
 import android.content.Intent
 import androidx.navigation.NavController
 import mozilla.components.lib.crash.Crash
-import org.mozilla.fenix.NavGraphDirections
+import mozilla.components.lib.crash.Crash.UncaughtExceptionCrash
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.appstate.AppAction
+import mozilla.components.lib.crash.CrashReporter
 
 /**
- * When the app crashes, the user has the option to report it.
- * Reporting fires an intent to the main activity which is handled here.
+ * Process the [Intent] from [CrashReporter] through which the app is informed about
+ * recoverable from native crashes.
  */
-class CrashReporterIntentProcessor : HomeIntentProcessor {
+class CrashReporterIntentProcessor(private val store: AppStore) : HomeIntentProcessor {
 
     override fun process(intent: Intent, navController: NavController, out: Intent): Boolean {
         return if (Crash.isCrashIntent(intent)) {
-            openToCrashReporter(intent, navController)
+            (Crash.fromIntent(intent) as? UncaughtExceptionCrash)?.let {
```

Comment:
Actually should this be `NativeCodeCrash`?  When I made my suggested I was looking at your cod using `UncaughtExceptionCrash`.  But shouldn't this be a non-fatal native crash?

Reply:
nit: do we need to log if a fatal/non-native crash arrives here? (should not happen)

Reply:
Indeed, this was initially ment for `NativeCodeCrash`es and the documentation of the class says so.
Changed accordingly.
Added an error log.